pyproject.toml

```
@@ -94,11 +94,43 @@ h5py = ">=3.2.1, <4.0.0"  # from geoh5py
 Pillow = ">=10.3.0, <10.4.0"  # from geoh5py
 pydantic = ">=2.5.2, <3.0.0"  # from geoh5py, curve-apps, geoapps-utils
 
-[tool.isort]
-profile = "black"
+[tool.ruff]
+target-version = "py310"
+
+[tool.ruff.lint]
+ignore = [
+    "B028",  # no-explicit-stacklevel for warnings.warn()
+    "E501",  # line-too-long - code is reformatted (do not care about comments and docstring)
+    "F401",  # unsused-import - covered by pycln
+    "RUF012", # TO DO
+    "UP038", # TO DO
+    "RUF015", # TO DO
```

"TODO" is a special comment marker highlighted in IDE.
Also used as the default keyword by some tools to extract and track TODOs (as in Jenkins)

```suggestion
    "RUF012",  # TODO
    "UP038",  # TODO
    "RUF015",  # TODO
```